### ISSUE TITLE
Max Grad Norm doesn't seem to behave right

### ISSUE THREAD

**Issue by LaRiffle** [pytorch/opacus#306]
## 🐛 Bug

Looking at a simple demo https://github.com/pytorch/opacus/blob/main/examples/cifar10.py, my intuition is that setting the `max_grad_norm` parameter high should result in no gradient clipping and hence should not decrease the accuracy.

However, just running this tutorial with a high parameter show it doesn't behave like this. Any reason why ?

```
python cifar10.py --workers 0 
```
Accuracy after epoch 3: `Test set:Loss: 1.903411 Acc@1: 0.335742 ` 

```
python cifar10.py --workers 0 --max-per-sample-grad_norm 1000
```
Accuracy after epoch 3: `Test set:Loss: 1483.235467 Acc@1: 0.098926 ` 

## Environment

``` 
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.5.2 (x86_64)
GCC version: Could not collect
Clang version: 12.0.5 (clang-1205.0.22.9)
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.12 (default, Aug 31 2021, 04:09:21)  [Clang 12.0.5 (clang-1205.0.22.9)] (64-bit runtime)
Python platform: macOS-11.5.2-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.4
[pip3] torch==1.10.0
[pip3] torchvision==0.11.1
[conda] Could not collect
```

**Comment by gkaissis** [comment]
Hey Theo! 
Did you turn the noise to very low when setting the clip norm to 1000? (Like 1e-12). Otherwise, you're dragging the noise up by 1e3 when you're increasing the noise multiplier.

**Comment by LaRiffle** [comment]
Ah yes, indeed I read in the doc:
```
Add Gaussian noise to ``p.grad`` calibrated to a given noise multiplier 
and max grad norm limit (``std = noise_multiplier * max_grad_norm``).
```
So it makes a lot of sense!
Thanks @gkaissis 🙏
